### PR TITLE
docs: document methods and add mypy

### PR DESCRIPTION
**Pull request by dtgoitia**

### CODE REVIEW

File: abot/bot.py [hunk 1]
```diff
@@ -26,15 +26,33 @@ class Abort(Exception):
...
+        and store/handle them before the Bot initializates the Backend.
+        """
         raise NotImplementedError
 
     async def initialize(self):
+        """Execute one time operation which should run when the Backend starts."""
         raise NotImplementedError()
 
     async def consume(self):
+        """Execute Backend logic."""
```
Comment (txomon) [inline abot/bot.py]: Please make note this is supposed to be an async generator that yields `Event` subtypes

File: abot/bot.py [hunk 2]
```diff
@@ -26,15 +26,33 @@ class Abort(Exception):
...
+        """Return whatever entity represents the current backend.
+
+        The entity returned by this method will be use to identify the current
+        backend and match incoming Events and MessageEvents to the current
+        backend. Bear in mind that multiple backends of the same type could be
+        running at the same time. Thus, this entity should be unique for
+        each backend instance.
+
+        The idea is to leave up to the bot author to decide which entity is
+        more convenient to identify the backend.
```
Comment (txomon) [inline abot/bot.py]: Put an example that this can be the `Entity` object that represents the bot's user

**Review by txomon — CHANGES_REQUESTED** [review]
LGTM, just a couple of small fixes

File: abot/bot.py (hunk 2 above)
Comment (dtgoitia) [reply abot/bot.py]: See line 60 for example.

File: abot/bot.py (hunk 1 above)
Comment (dtgoitia) [reply abot/bot.py]: Done

**Review by txomon — APPROVED** [review]
Nice, thank you!

**Pull request merged**